api/src/main/java/org/apache/iceberg/SnapshotUpdate.java

```diff
@@ -71,4 +71,32 @@ default ThisT toBranch(String branch) {
             "Cannot commit to branch %s: %s does not support branch commits",
             branch, this.getClass().getName()));
   }
+
+  /**
+   * Enables snapshot validation with a user-provided function, which must throw a {@link
+   * org.apache.iceberg.exceptions.ValidationException} on validation failures.
```

Comment:
I think this should document how the validator is called. Is the latest snapshot first or last? I think that affects how validations are written so it seems important.